packages/royalties/contracts/MultiRoyaltyDistributer.sol

```diff
@@ -0,0 +1,166 @@
+abstract contract MultiRoyaltyDistributer is IEIP2981, IMultiRoyaltyDistributer, ERC165Upgradeable {
+    uint16 internal constant TOTAL_BASIS_POINTS = 10000;
+    uint16 public _defaultRoyaltyBPS;
+    address payable public _defaultRoyaltyReceiver;
+    address public royaltyManager;
+
+    mapping(uint256 => address payable) public _tokenRoyaltiesSplitter;
+    uint256[] private _tokensWithRoyalties;
+
+    function __MultiRoyaltyDistributer_init(
+        address payable defaultRecipient,
+        uint16 defaultBps,
+        address _royaltyManager
+    ) internal {
+        _defaultRoyaltyReceiver = defaultRecipient;
+        _defaultRoyaltyBPS = defaultBps;
+        royaltyManager = _royaltyManager;
+    }
+
+    /// @notice EIP 165 interface funtion
+    /// @dev used to check interface implemented
+    /// @param interfaceId to be checked for implementation
+    function supportsInterface(bytes4 interfaceId)
+        public
+        view
+        virtual
+        override(ERC165Upgradeable, IERC165)
+        returns (bool)
+    {
+        return
+            interfaceId == type(IEIP2981).interfaceId ||
+            interfaceId == type(IEIP2981MultiReceiverRoyaltyOverride).interfaceId ||
+            super.supportsInterface(interfaceId);
+    }
+
+    /// @notice sets token royalty
+    /// @dev deploys a splitter if creator doen't have one
+    /// @param tokenId id of token
+    /// @param royaltyBPS the bps of for EIP2981 royalty
+    /// @param creator of the token
+    function _setTokenRoyalties(
+        uint256 tokenId,
+        uint16 royaltyBPS,
+        address payable recipient,
+        address creator
+    ) internal {
+        require(royaltyBPS < TOTAL_BASIS_POINTS, "Invalid bps");
+        address payable creatorSplitterAddress = IRoyaltyManager(royaltyManager).deploySplitter(creator, recipient);
+        _tokenRoyaltiesSplitter[tokenId] = creatorSplitterAddress;
+        _tokensWithRoyalties.push(tokenId);
+        emit TokenRoyaltySet(tokenId, royaltyBPS, recipient);
+    }
+
+    /**
+     * @dev Sets default royalty. When you override this in the implementation contract
+     * ensure that you access restrict it to the contract owner or admin
+     */
+    function _setDefaultRoyaltyBps(uint16 bps) internal {
+        require(bps < TOTAL_BASIS_POINTS, "Invalid bps");
+        _defaultRoyaltyBPS = bps;
+        emit DefaultRoyaltyBpsSet(bps);
+    }
+
+    /**
+     * @dev Sets default royalty. When you override this in the implementation contract
+     * ensure that you access restrict it to the contract owner or admin
+     */
+    function _setDefaultRoyaltyReceiver(address payable defaultReceiver) internal {
+        require(defaultReceiver != address(0), "Default receiver can't be zero");
+        _defaultRoyaltyReceiver = defaultReceiver;
+        emit DefaultRoyaltyReceiverSet(defaultReceiver);
+    }
+
+    /**
+     * @dev See {IEIP2981MultiReceiverRoyaltyOverride-getTokenRoyalties}.
```

Comment:
These dev comments are they copied from Manifold? If they are then they should be updated to say so

Reply:
I have to check it out